src/visualization/components/internal/TimeTickInput.tsx

```diff
@@ -135,10 +170,15 @@ export const TimeTickInput: FC<TimeTickInputProps> = props => {
   }
 
   const handleSelectDate = (date: string) => {
-    const dateRFC3339 = new Date(date).valueOf()
-    setTickOptionInput(dateRFC3339 === dateRFC3339 ? String(dateRFC3339) : '')
-    setTickOptionInputStatus(ComponentStatus.Default)
-    updateTickOption(dateRFC3339)
+    if (isValidRFC3339(date)) {
+      const dateRFC3339 = convertDateToLocalTime(new Date(date))
```

Comment:
I think this looks good for when the time zone is set as `local` in our app, I'm not sure if this is addressing the UTC timezone? 

From my interpretation, the line 174 will always convert it to local timezone. I could be missing something tho.

Reply:
For example, if the app timezone is set to local, we'd see the input string formatted in the right format, but I think when the app is in `UTC` we would want the input string that is displayed, in UTC format? 

That's my only concern.

Reply:
Ah, good call. I will add that in. We will convert it to the chosen `timeZone`